File: 1회차/programmers_algorithm/3.cpp
#include <string>
#include <vector>
#include <set>
#include <algorithm>

using namespace std;

int solution(string numbers) {
    int n = numbers.size();
    vector<int> arr;
    for (int i = 0; i < n; i++) {
        arr.push_back(numbers[i]-'0');
    }
    sort(arr.begin(),arr.end());
    set<int> s;
    do {
        int num = 0;
        for (int i = 0; i < n; i++) {
            num=num*10+arr[i];
            int tag=1;
            for (int i = 2; i*i <= num; i++) {
                if (num%i==0) {tag=0; break;}
            }
            if (tag && num>1) s.insert(num);
        }
    } while (next_permutation(arr.begin(),arr.end()));
    return s.size();
}

File: 1회차/programmers_algorithm/3.py
from itertools import permutations
import math
def solution(numbers):
    n = len(numbers)
    arr = []
    for i in range(n):
        arr.append(int(numbers[i]))
    arr.sort()
    
    p = list(permutations(arr,len(arr)))
    s = {}
    for i in range(len(p)):
        num=0
        for j in range(n):
            num=num*10+p[i][j]
            tag=1
            for k in range(2,int(math.sqrt(num))+1):
                if num%k==0:
                    tag=0
                    break
            if tag and num>1:
                s[num]=1

    return len(s)